### PR TITLE
Detect a return value of Ref<T> & RefPtr<T>

### PR DESCRIPTION
**Pull request by rniwa**
This PR makes the checker not emit warning when a function is called with a return value of another function when the return value is of type Ref<T> or RefPtr<T>.

### CODE REVIEW

File: clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp [hunk 1]
```diff
@@ -118,6 +118,26 @@ bool isCtorOfRefCounted(const clang::FunctionDecl *F) {
...
 
+bool isReturnValueRefCounted(const clang::FunctionDecl *F) {
+  assert(F);
+  auto *type = F->getReturnType().getTypePtrOrNull();
+  while (type) {
+    if (auto *elaboratedT = dyn_cast<ElaboratedType>(type)) {
+      type = elaboratedT->desugar().getTypePtrOrNull();
+      continue;
+    }
+    if (auto *specialT = dyn_cast<TemplateSpecializationType>(type)) {
```
Comment (haoNoQ) [inline clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp:129]: `.getTypePtrOrNull()` can be dropped here as well:
```suggestion
  QualType type = F->getReturnType();
  while (!type.isNull()) {
    if (auto *elaboratedT = type->getAs<ElaboratedType>()) {
      type = elaboratedT->desugar();
      continue;
    }
    if (auto *specialT = type->getAs<TemplateSpecializationType>()) {
```

Comment (rniwa) [reply clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp]: Sounds good. Fixed.

**Review by haoNoQ — APPROVED** [review]
LGTM!

**Comment by haoNoQ** [comment]
Ok this needs a rebase after the other two PRs landed.

**Pull request merged**